sycl/include/CL/sycl/group.hpp

```diff
@@ -120,25 +121,134 @@ template <int Dimensions = 1> class group {
     return globalRange[dimension];
   }
 
+  id<Dimensions> get_local_id() const {
+#ifdef __SYCL_DEVICE_ONLY__
+    return __spirv::initLocalInvocationId<Dimensions, id<Dimensions>>();
+#else
+    throw runtime_error("get_local_id() is not implemented on host device",
+                        PI_INVALID_DEVICE);
+    // Implementing get_local_id() on host device requires ABI breaking change.
+    // It requires extending class group with local item which represents
+    // local_id. Currently this local id is only used in nd_item and group
+    // cannot access it.
+#endif
+  }
+
+  template <int dims = Dimensions>
+  typename detail::enable_if_t<(dims == 1), size_t>
+  get_local_linear_id() const {
+    id<Dimensions> localId = get_local_id();
+    return localId[0];
+  }
```

Comment:
Yes, it does seem weird that we would implement this function as a template when it is not that way in the spec.  It seems like it would be easy to fix by calling some separate templated function as John proposes.  I assume the separate function could also be a private member function of `group`?

```
class group {
 public:
  size_t get_local_linear_id() const {
    return get_local_linear_id_impl<Dimensions>();
  }

private:
  template <int dims>
  typename detail::enable_if_t<(dims == 1), size_t>
  get_local_linear_id_impl() const {
    id<Dimensions> localId = get_local_id();
    return localId[0];
  }
}
```

Or, you could use `constexpr-if`, but that would require the compiler to be C++17.

Reply:
Done